lib/matplotlib/tests/test_colors.py

```diff
@@ -44,6 +45,32 @@ def test_resample():
     assert_array_almost_equal(lc3([0, 0.5, 1]), expected)
 
 
+def test_colormap_copy():
+    cm = plt.cm.Reds
+    cm([-1, 0, .5, np.nan, np.inf])
+    cm.set_bad('y')
+    cm2 = copy.copy(cm)
+    cm2.set_bad('b')
+    cm.set_under('k')
+    cm2.set_under('r')
+    cm.set_over('c')
+    cm2.set_over('m')
+
+    expected1 = plt.cm.Reds
+    expected2 = plt.cm.Reds
+    expected1([-1, 0, .5, np.nan, np.inf])
+    expected2([-1, 0, .5, np.nan, np.inf])
+    expected1.set_bad('y')
+    expected2.set_bad('b')
+    expected1.set_under('k')
+    expected2.set_under('r')
+    expected1.set_over('c')
+    expected2.set_over('m')
+
+    assert_array_equal(cm._lut, expected1._lut)
```

Comment:
If I do the following : 
```python
cm = plt.cm.Reds
ret1 = cm([-1, 0, 0.5, 1, np.nan, np.inf])
cm2 = copy.copy(cm)
cm2.set_bad('g')
ret2 = cm([-1, 5, .5, 1, np.nan, np.inf])
assert_array_equal(ret1, ret2)
```
I get the assertion error.
The copy function does work but how do I find the error?